src/Backoffice/Flight/App/Requests/UpsertFlightRequest.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Lightit\Backoffice\Flight\App\Requests;
+
+use Carbon\CarbonImmutable;
+use Illuminate\Foundation\Http\FormRequest;
+use Illuminate\Validation\Rule;
+use Illuminate\Validation\Validator;
+use Lightit\Backoffice\Airline\Domain\Models\Airline;
+use Lightit\Backoffice\City\Domain\Models\City;
+use Lightit\Backoffice\Flight\Domain\DataTransferObject\FlightDto;
+
+class UpsertFlightRequest extends FormRequest
+{
+    public const AIRLINE_ID = 'airline_id';
+
+    public const DEPARTURE_CITY_ID = 'departure_city_id';
+
+    public const ARRIVAL_CITY_ID = 'arrival_city_id';
+
+    public const DEPARTURE_TIME = 'departure_time';
+
+    public const ARRIVAL_TIME = 'arrival_time';
+
+    /**
+     * @return array<string, mixed>
+     */
+    public function rules(): array
+    {
+        return [
+            self::AIRLINE_ID => ['required', Rule::exists(Airline::class, 'id')],
+            self::DEPARTURE_CITY_ID => ['required', Rule::exists(City::class, 'id')],
+            self::ARRIVAL_CITY_ID => ['required', Rule::exists(City::class, 'id')],
+            self::DEPARTURE_TIME => ['required', 'date_format:Y-m-d H:i:s'],
+            self::ARRIVAL_TIME => ['required', 'date_format:Y-m-d H:i:s'],
+        ];
+    }
+
+    public function after(): array
+    {
+        return [
+            function (Validator $validator): void {
+                if ($this->validateSameOriginAndDestinationCity()) {
+                    $validator->errors()->add(
+                        self::ARRIVAL_CITY_ID,
+                        'The arrival city cannot be the same as the departure city.'
+                    );
+                }
+
+                if (! $this->validateFlightDateTimes()) {
+                    $validator->errors()->add(
+                        self::DEPARTURE_TIME,
+                        'The departure time must be before the arrival time.'
+                    );
+                }
+            },
+        ];
+    }
```

Comment:
I know it serves the same purpuse, but i would run this before, not after the validations, since the validations check in the DB, doing it before saves us doing the queries